lib/simulator.py

```diff
@@ -241,11 +273,21 @@ def on_size(self, e):
         setsize_window_width = self.width * self.scale + decorations_width + self.margin * 2
         setsize_window_height = (self.height) * self.scale + decorations_height + self.margin * 2
 
+        # set minimum width (force space for control buttons)
         if setsize_window_width < 600:
             setsize_window_width = 600
 
         self.SetSize(( setsize_window_width, setsize_window_height))
 
+        # center the simulation on screen if not called by params
+        # else center horizontally
+        if self.center_window == True:
+            self.Centre()
+        else:
+            current_pos = self.GetPosition()
+            display_height = wx.ClientDisplayRect()
+            self.SetPosition((current_pos[0], display_height[3] / 2 - setsize_window_height / 2))
```

Comment:
Oof, I didn't understand things fully when I suggested the `center_window` argument.  I forgot that Params wants to set the X part of the position but it can't set the Y since it doesn't know how tall the simulator will be.

I know I was the one that suggested `center_window`, but now I'm having second thoughts.  Maybe instead of that, we could have the EmbroiderySimulator take a `x_position` argument?  Then instead of using `current_pos[0]` here, it would just use `self.x_position`.  If the caller didn't specify an `x_position`, then we center the window.  It's basically the same thing we have now, but to me it seems a little less confusing.